### ISSUE TITLE
Load and Save Sessions from/to Network

### ISSUE THREAD

**Issue by jimradford** [jimradford/superputty#518]
Pondering a plugin architecture to allow saving session data to the network. This would allow those that use the application from multiple computers to keep session data in sync between computers.

Initially I think I will create a simple PHP application and associated plugin to self host the sessions data. 

In the future it would be nice to have support for some of the cloud providers that exist, AWS, Azure, Google off the top of my head.

**Comment by brnl** [comment]
I have the full installation (in this case, the extracted zip) running on Owncloud (and Dropbox before). that works fine if you make sure the paths to putty are relative in the settings. Works fine on multiple computers.

**Comment by joeyhub** [comment]
I am already working on something in this direction.

The problem is that the codebase is not really well suited to this. You can tell that it was made in a bit of a rush and lots of things were done the easiest way for the requirements at the time.

The SessionData is the biggest problem at the moment. I am already about 75% of the way through rewriting this in my local edition.

I am separating it into its own model and changing the structure so that it accurately represents a tree.

**Comment by waukeerules** [comment]
Would this resolve load times when refreshing the session list? I currently have about 1,000 sessions stored in my xml instance. And it takes quite a bit of time to load all those sessions.

**Comment by jimradford** [comment]
@waukeerules No, This won't fix load times for sessions, the issue there is the treeview control itself needs to be changed to. Likely a delayed load or possibly a virtual data source might be the solution.
